compiler/AST/symbol.cpp

```diff
@@ -472,12 +472,10 @@ const char* Symbol::getSanitizedMsg(std::string msg) const {
   // TODO: Support explicit title and reference targets like in reST direct hyperlinks (and having only target
   //       show up in sanitized message).
   // TODO: Allow prefixing content with ! (and filtering it out in the sanitized message)
-  // TODO: Allow prefixing content with ~ (and having it only display last component of target)
-  static const auto reStr = R"(\B\:(mod|proc|iter|data|const|var|param|type|class|record|attr|enum)\:`([!$\w\$\.]+)`\B)";
+  static const auto reStr = R"(\B\:(mod|proc|iter|data|const|var|param|type|class|record|attr|enum)\:`((?:[!$\w\$\.]+)|(?:~[$\w\$]+\.?))+`\B)";
```

Comment:
Reading the regexp I think this could match some invalid cases where `~` appears but isn't the first character like:

```
:proc:`a~b`
```

Though I think we'd have the same issue existing with with `!`.

I'm not sure I care that much about it though given that `~` isn't a legal identifier character in Chapel and the regexp isn't really meant to validate the legality of the string so much as apply sensible filtering when producing a warning message.

So I'd say if you can figure out a way to make it not match on cases like that, that'd be great, but I wouldn't sweat over it if it's not that straightforward.